src/main/java/org/quiltmc/loader/impl/gui/QuiltGuiEntry.java

```diff
@@ -109,4 +161,74 @@ private static void addThrowable(QuiltStatusNode node, Throwable e, Set<Throwabl
 			addThrowable(sub, t, seen);
 		}
 	}
+
+	private static void fork(QuiltStatusTree tree, boolean shouldWait) throws Exception {
+		List<String> commands = new ArrayList<>();
+		commands.add(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java");
```

Comment:
How does vanilla launcher handle this? On windows at least the launcher will use the java executable in launcher folder, so this may fail if on macOS forking the JVM does not have a path set BUT a different jvm launched it.

Reply:
Why would that affect this?

According to `java.lang.System.getProperties()`:
> java.home | Java installation directory

Which, to me, indicates we'd always fork with whatever JVM was used to run minecraft. (As for how jvms are organised... I have no idea, I'm really just assuming that this `<java-home>/bin/java` is always valid)

Reply:
Ah didn't realize that.